### ISSUE TITLE
[backport] [UI] In mcapp answers keys included dots don't apply values but make them empty

### ISSUE THREAD

**Issue by codyrancher** [rancher/rancher#24025]
Backport of rancher/rancher#23379

**Comment by westlywright** [comment]
Available in `latest-2.3`.

**Comment by soumyalj** [comment]
Tested with v2.3-head with UI index latest-2.3

Steps:
Add a catalog: URL: https://github.com/helm/charts.git
Launch a mcapp - Stolon
Add answers:
test1: value1
test2.key: value2
test3: value3
test4.key: value4
In the http request, the UI sent the below answers as expected:
```
answers: [{type: "answer", clusterId: null, projectId: null,…}]
0: {type: "answer", clusterId: null, projectId: null,…}
clusterId: null
projectId: null
type: "answer"
values: {test1: "value1", test2.key: "value2", test3: "value3", test4.key: "value4"}
test1: "value1"
test2.key: "value2"
test3: "value3"
test4.key: "value4"

```
View in API:
```
"answers": [
{
"clusterId": null,
"projectId": null,
"type": "/v3/schemas/answer",
"values": {
"test1": "value1",
"test2.key": "value2",
"test3": "value3",
"test4.key": "value4"
}
}
]


```
